test-utils/test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/matchers/HandlerClassMatcherTest.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.justice.services.test.utils.core.matchers;
+
+import static org.junit.Assert.assertThat;
+import static uk.gov.justice.services.core.annotation.Component.COMMAND_API;
+import static uk.gov.justice.services.core.annotation.Component.QUERY_API;
+import static uk.gov.justice.services.test.utils.core.matchers.HandlerClassMatcher.isHandlerClass;
+import static uk.gov.justice.services.test.utils.core.matchers.HandlerMatcher.isHandler;
+import static uk.gov.justice.services.test.utils.core.matchers.HandlerMethodMatcher.method;
+
+import uk.gov.justice.services.test.utils.core.matchers.ServiceComponentTestClasses.NoHandlerMethod;
+import uk.gov.justice.services.test.utils.core.matchers.ServiceComponentTestClasses.NoServiceComponentAnnotation;
+import uk.gov.justice.services.test.utils.core.matchers.ServiceComponentTestClasses.ValidCommandApi;
+
+import org.junit.Test;
+
+public class HandlerClassMatcherTest {
+
+    @Test
+    public void shouldMatchIfServiceCompontAnnotation() throws Exception {
+        assertThat(ValidCommandApi.class, isHandlerClass(COMMAND_API));
+    }
+
+    @Test(expected = AssertionError.class)
+    public void shouldNotMatchIfNoServiceComponentAnnotation() throws Exception {
+        assertThat(NoServiceComponentAnnotation.class, isHandlerClass(COMMAND_API));
+    }
+
+    @Test(expected = AssertionError.class)
+    public void shouldNotMatchInstanceOfAHandlerWithTheWrongAnnotation() throws Exception {
+        assertThat(ValidCommandApi.class, isHandler(QUERY_API));
+    }
+
+    @Test
+    public void shouldMatchHandlerMethod() throws Exception {
+        assertThat(ValidCommandApi.class, isHandlerClass(COMMAND_API).with(method("testA").thatHandles("testA")));
+    }
+
+    @Test(expected = AssertionError.class)
+    public void shouldNotMatchWhenNoHandlerMethod() throws Exception {
+        assertThat(NoHandlerMethod.class, isHandlerClass(COMMAND_API).with(method("testA").thatHandles("testA")));
```

Comment:
maybe update the handles string to context.commandA or something to make it clear that it is a handlesAnnotation type thing. for these 2 tests